core/trino-main/src/main/java/io/trino/operator/aggregation/state/LongDecimalWithOverflowAndLongStateSerializer.java

```diff
@@ -42,7 +42,14 @@ public void serialize(LongDecimalWithOverflowAndLongState state, BlockBuilder ou
             long overflow = state.getOverflow();
             long[] decimal = state.getDecimalArray();
             int offset = state.getDecimalArrayOffset();
-            VARBINARY.writeSlice(out, Slices.wrappedLongArray(count, overflow, decimal[offset], decimal[offset + 1]));
+            long[] buffer = new long[4];
+            buffer[0] = decimal[offset + 1];
+            buffer[1] = decimal[offset];
+            buffer[2] = overflow;
+            buffer[3] = count;
+            int bufferLength = 1 + (decimal[offset] == 0 ? 0 : 1);
```

Comment:
bufferLength -> decimalsCount ?

Reply:
I meant that only for the `1 + (decimal[offset] == 0 ? 0 : 1)` part, the variable on next line should definitely be named `bufferLength`.

Reply:
that's actually way better. Renamed.